clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -2953,7 +2973,8 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t outp
             input->needs_init = false;
         }
         if (!input->queue.empty()) {
-            record = input->queue.front();
+            input->cur_queue_record = input->queue.front();
+            record = input->cur_queue_record.record;
```

Comment:
Please make a helper function as this looks too easy to make an error and not call both of these every time the front of the queue is examined.